src/dex/gmx/pool.ts

```diff
@@ -97,6 +105,27 @@ export class GMXEventPool extends ComposedEventSubscriber<PoolState> {
     return onChainState;
   }
 
+  async getMaxAmountIn(_tokenIn: Address, _tokenOut: Address): Promise<bigint> {
+    const cacheKey = `maxAmountIn_${_tokenIn}_${_tokenOut}`;
+    const maxAmountCached = await this.dexHelper.cache.get(
+      this.parentName,
+      this.network,
+      cacheKey,
+    );
+    if (maxAmountCached) return BigInt(maxAmountCached);
+    const maxAmount: string = await this.reader.methods
+      .getMaxAmountIn(this.vault.vaultAddress, _tokenIn, _tokenOut)
+      .call();
+    this.dexHelper.cache.setex(
```

Comment:
minor but this isn't `await` -ed

Reply:
shouldn't await setting stuff in the cache really, there's no reason to wait for it